File: gitdot-web/packages/api/src/endpoint/review/publish-review.ts
import { z } from "zod";
import { ReviewResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const DiffUpdate = z.object({
  position: z.number().int(),
  message: z.string().optional(),
});
export type DiffUpdate = z.infer<typeof DiffUpdate>;

export const PublishReviewRequest = z.object({
  title: z.string().optional(),
  description: z.string().optional(),
  diffs: z.array(DiffUpdate).optional(),
});
export type PublishReviewRequest = z.infer<typeof PublishReviewRequest>;

export const PublishReviewResponse = ReviewResource;
export type PublishReviewResponse = z.infer<typeof PublishReviewResponse>;

export const PublishReview = {
  path: "/repository/{owner}/{repo}/review/{number}/publish",
  method: "POST",
  request: PublishReviewRequest,
  response: PublishReviewResponse,
} as const satisfies Endpoint;
export type PublishReview = typeof PublishReview;

File: gitdot-api/src/endpoint/review/publish_review.rs
use serde::{Deserialize, Serialize};

use crate::endpoint::Endpoint;

pub struct PublishReview;

impl Endpoint for PublishReview {
    const PATH: &'static str = "/repository/{owner}/{repo}/review/{number}/publish";
    const METHOD: http::Method = http::Method::POST;

    type Request = PublishReviewRequest;
    type Response = ();
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct PublishReviewRequest {}
